### PR TITLE
Move ItemContext to react root and fix imports

### PR DESCRIPTION
**Pull request by marcosewbank**
#### What problem is this solving?

This change will allow import the context to customized components.

#### How should this be manually tested?

With this change you can import the context like this:

```
import { useItemContext } from 'vtex.product-list/ItemContext'

const { item, loading, onQuantityChange } = useItemContext()
```

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
_ | New feature 
_ | Breaking change 
✔️ | Technical improvements

### CODE REVIEW

**Review by marcoskwkm — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+- Move `ItemContext` to react folder.
```
Comment (marcoskwkm) [inline CHANGELOG.md]: Add a `### Changed` header, and change this to the past tense.

**Review by darlenedms — CHANGES_REQUESTED** [review]
Same as @marcoskwkm

**Review by kaisermann — CHANGES_REQUESTED** [review]
Same as @marcoskwkm. Check the [keep a changelog](https://keepachangelog.com/en/1.0.0/) documentation to see its guidelines 😁

**Comment by marcosewbank** [comment]
@kaisermann @darlenedms @marcoskwkm Changelog fixed ^^

**Review by victorhmp — APPROVED** [review]

File: react/ItemContext.tsx [hunk 2]
```diff
@@ -27,3 +27,5 @@ export const useItemContext = () => {
 
   return context
 }
+
+export default { useItemContext }
```
Comment (kaisermann) [inline react/ItemContext.tsx:31]: The `useItemContext` is already a named export (see the `export const useItemContext` just above). Why are you exporting it as the default export as well?

**Review by kaisermann — CHANGES_REQUESTED** [review]

File: react/ItemContext.tsx (hunk 2 above)
Comment (lucasecdb) [reply react/ItemContext.tsx:31]: because of VTEX IO imports. If you only export it as a named export, the import in other apps won't work

Comment (kaisermann) [reply react/ItemContext.tsx:31]: Oh, right, got it!

**Review by kaisermann — APPROVED** [review]

**Review by darlenedms — APPROVED** [review]

**Pull request merged**